feature/bgp/gracefulrestart/ate_tests/err/README.md

```diff
@@ -0,0 +1,289 @@
+  3. Upon multiple failures on the Speaker side resulting in GR restart timer or the stale path timer not to expire on the helper side.
+  4. Upon expiration of the stale path timer
+Under the aforementioned conditions, the routes received from the neighbor under failure must be held for a configurable duration and processed through an additional configurable routing policy while being held in a “stale” state.
+
+Since the route retention is purely local action of the receiving speaker, this action should not require any additional capabilities advertisements beyond capability 64 (Graceful Restart), and should not be confused with or require capability 71 (Long-Lived Graceful Restart) from the sending speaker.
+
+**How is this different from LLGR as tested in RT-1.14?**
+As per the [IETF Draft on LLGR](https://tools.ietf.org/html/draft-ietf-idr-long-lived-gr), we have the following that is different from EER.
+
+  - Section 4.2 / 4.3 of the draft: mandates what communities are in use and what their specific behavior should be. For example: "The "LLGR_STALE" community must be advertised by the GR helper and also MUST NOT be removed by other receiving peers." and anyone that receives that route MUST treat the route as least-preferred. This isnt the case for ERR. There arent any communities attached to Stale routes thereby mandating their depreference.
+  - Section 4.7: Different conditions for partial deployment of LLGR is a no-op for ERR as it builds on the concepts of RFC8538 and hence there arent any special communities expected to be sent or received for the stale routes.
+
+## Testbed type
+[atedut_2.testbed](https://github.com/openconfig/featureprofiles/blob/main/topologies/atedut_2.testbed)
+
+## Procedure
+
+* Test environment setup
+  ## Topology
+Create the following connections:
+```mermaid
+graph LR; 
+A[ATE:Port1] <-- IBGP(ASN100) --> B[Port1:DUT:Port2];
+B <-- EBGP(ASN200) --> C[Port2:ATE];
+```
+* ATE:Port1 runs IBGP and must advertise the following IPv4 and IPv6 prefixes with the corresponding commiunitty attributes
+  * IPv4Prefix1 and IPv6Prefix1 with community NO-ERR
+  * IPv4Prefix2 and IPv6Prefix2 with community ERR-NO-DEPREF
+  * IPv4Prefix3 and IPv6Prefix3 with community TEST-IBGP
+* ATE:Port2 runs EBGP and must advertise the following IPv4 and IPv6 prefixes with the corresponding commiunitty attributes
+  * IPv4Prefix4 and IPv6Prefix4 with community NO-ERR
+  * IPv4Prefix5 and IPv6Prefix5 with community ERR-NO-DEPREF
+  * IPv4Prefix6 and IPv6Prefix6 with community TEST-EBGP
+* DUT has the following configuration on its IBGP and EBGP peering
+  * Extended route retention (ERR) enabled.
+  * ERR configuration has the retention time of 300 secs configured
+  * ERR has a retention-policy `STALE-ROUTE-POLICY` attached. The DUT **MUST** apply this policy on post-policy adj-rib-in routes. Therefore, import export policy applied to the routes must not be overridden by this policy, it MUST instead be additive.
```

Comment:
Can you define somewhere what the retention policy is defined to be and how it applies to routes? Is it applied by default to all sessions?

I think you are saying that there's a specific order of operations for these policies -- and they are NOT part of the normal import/export policy chain. Is that right?

Reply:
I had a section under Procedure explaining how the ERR policy differentiates from the regular import/export policy. However, I now consolidated and added more details under the summary section w/ a new subsection for ERR policy. PTAL